### PR TITLE
Add support for background-position in geckolib.

### PR DESCRIPTION
**Pull request by heycam**
r? @Manishearth 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because this is a geckolib-only change

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12651)

### CODE REVIEW

File: components/style/gecko_conversions.rs [hunk 1]
```diff
@@ -34,3 +34,41 @@ impl From<nsStyleCoord_CalcValue> for CalcLengthOrPercentage {
...
+    fn from(other: nsStyleCoord_CalcValue) -> LengthOrPercentage {
+        match (other.mHasPercent, other.mLength) {
+            (false, _) => LengthOrPercentage::Length(Au(other.mLength)),
+            (true, 0) => LengthOrPercentage::Percentage(other.mPercent),
+            _ => {
+                let calc = CalcLengthOrPercentage {
+                    length: Some(Au(other.mLength)),
+                    percentage: None,
+                };
+                LengthOrPercentage::Calc(calc)
```
Comment (Manishearth) [inline components/style/gecko_conversions.rs]: this can just be LoP::Calc(other.into())

Comment (Manishearth) [inline components/style/gecko_conversions.rs]: Also, why is percentage none here? What about the case when calc is a length and a percentage?

Comment (heycam) [inline components/style/gecko_conversions.rs]: My code is also wrong, I see (should be `other.mPercent`, not `None`, for the percentage value).

**Pull request merged**